qiskit_nature/operators/second_quantization/fermionic_op.py

```diff
@@ -346,6 +346,10 @@ def __repr__(self) -> str:
             ")"
         )
 
+    def __iter__(self) -> Iterable[tuple[list[tuple[str, int]], complex]]:
```

Comment:
I think we usually use `Tuple` and `List` for type hints which come from the Typing module.

Reply:
That's true, but @ikkoham changed that in this file in https://github.com/Qiskit/qiskit-nature/pull/506 so I did it this way to be consistent. I don't know why @ikkoham changed it.

Reply:
The lowercase variants, i.e. using tuple etc is a change for 3.9 onwards https://docs.python.org/3/whatsnew/3.9.html#type-hinting-generics-in-standard-collections which avoids duplicate similar names from Typing and confusion that results from using the different ones in the different contexts. This is better I think and I imagine we will transition the whole code base over. This has `from __future__ import annotations` to allow its use on earlier versions. I believe this became possible one we dropped 3.6 as supported where before we stuck to Typing.

Reply:
A minor point, but the return value of `__iter__` is `Iterator`, not `Iterable` - the sole difference being that `Iterator` has the `__next__` method as well.